eo-maven-plugin/src/test/java/org/eolang/maven/PullMojoTest.java

```diff
@@ -154,22 +162,92 @@ void pullsUsingOfflineHash(@TempDir final Path temp) throws IOException {
     void skipsPullMojo(@TempDir final Path temp) throws IOException {
         final FakeMaven maven = new FakeMaven(temp);
         maven.foreignTojos()
-            .add("org.eolang.io.stdout")
+            .add(PullMojoTest.STDOUT)
             .withScope("compile")
             .withVersion("*.*.*");
         maven.with("skip", true)
-            .with("objectionary", new Objectionary.Fake())
             .execute(PullMojo.class);
         MatcherAssert.assertThat(
-            new Home(temp.resolve("target")).exists(
-                Paths.get(
-                    String.format(
-                        "%s/org/eolang/io/stdout.eo",
-                        PullMojo.DIR
-                    )
-                )
-            ),
+            PullMojoTest.exists(temp, PullMojoTest.SOURCE),
             Matchers.is(false)
         );
     }
+
+    @Disabled
+    @Test
+    void pullsVersionedObjectSuccessfully(@TempDir final Path temp) throws IOException {
+        final FakeMaven maven = new FakeMaven(temp);
+        maven.foreignTojos()
+            .add("org.eolang.io.stdout|9c93528")
+            .withVersion("*.*.*");
+        maven.execute(PullMojo.class);
+        MatcherAssert.assertThat(
+            PullMojoTest.exists(temp, PullMojoTest.VERSIONED),
+            Matchers.is(true)
+        );
+    }
+
+    @Disabled
+    @Test
+    void pullsProbedVersionedObjectFromOneObjectionary(@TempDir final Path temp)
+        throws IOException {
+        final CommitHash hash = new ChCached(new CommitHashesMap.Fake().get("0.28.5"));
+        new FakeMaven(temp)
+            .with("withVersions", true)
+            .with("objectionaries", new Objectionaries.Fake(new OyRemote(hash)))
+            .withVersionedHelloWorld()
+            .execute(new FakeMaven.Pull());
+        MatcherAssert.assertThat(
+            PullMojoTest.exists(temp, PullMojoTest.VERSIONED),
+            Matchers.is(true)
+        );
+    }
+
+    @Disabled
+    @Test
+    void pullsProbedVersionedObjectsFromDifferentObjectionaries(@TempDir final Path temp)
+        throws IOException {
+        final Map<String, CommitHash> hashes = new CommitHashesMap.Fake();
+        final CommitHash first = hashes.get("0.28.5");
+        final CommitHash second = hashes.get("0.28.6");
+        final CommitHash third = hashes.get("0.28.7");
+        new FakeMaven(temp)
+            .with(
+                "objectionaries",
+                new ObjsDefault(
+                    new MapEntry<>(first, new OyRemote(first)),
+                    new MapEntry<>(second, new OyRemote(second)),
+                    new MapEntry<>(third, new OyRemote(third))
+                )
+            )
+            .with("withVersions", true)
+            .with("hsh", third)
+            .withVersionedProgram()
+            .execute(new FakeMaven.Pull());
+        MatcherAssert.assertThat(
```

Comment:
@maxonfjvipon Could you add assertion messages for all new tests, please?
https://www.yegor256.com/2023/01/19/layout-of-tests.html#assertions